### ISSUE TITLE
temps are stuck in PLA range.

### ISSUE THREAD

**Issue by Nathan22211** [sb53systems/G-Code-Flow-Temperature-Controller#15]
it seems the modified gcode doesn't account for nor offset for the minimum temp set in the slicer or the post-processor. I have the range set from 245-280 C and it's going down to 196C

![image](https://github.com/user-attachments/assets/77d70b83-8811-4b28-b7a4-a39114b20320)
LMK if it's an error on my part, I may be doing something wrong within orca slicer that causes this

**Comment by sb53systems** [comment]
@Nathan22211 You are supposed to use the lowest temperature for 1mm3/s Flow, and the Highest Temp for you max flow.  
The temperature is estimated by the script in a linear way between (Middle Temp to Low Temp ) and (Middle Temp to Max Temp), the script will keep decreasing or increase the temperature if the flow is lowest or heigher thant the indicated flow rate in the fields.  
      
![image](https://github.com/user-attachments/assets/b4705ab5-450c-408a-b9ac-5daea67c5e3f)  
  
I remain at your disposal if you need further information.  
  
Best reguards,  
Salim

**Comment by sb53systems** [comment]
I will add a calibration section later, which explains the calibration process with 3 to 5 prints,   
I would also add a youtube video with a theoretical part and a practical part.  
  
let me know if you need further information.  
  
Best reguards,  
Salim

**Comment by Nathan22211** [comment]
ok, well is the generated code meant to be in the negatives sometimes? 
![image](https://github.com/user-attachments/assets/e76b265e-ebea-47cd-b308-c4758abdfccb)

**Comment by sb53systems** [comment]
It's just a display bug present in the current version, look at the chart curve instead. The G-Code is correct.  
I recommend you to use 1mm3/s  as min Flow (Next version this value will be fixed).  
  
Salim.

**Comment by sb53systems** [comment]
> I will add a calibration section later, which explains the calibration process with 3 to 5 prints,  
  
https://github.com/sb53systems/G-Code-Flow-Temperature-Controller?tab=readme-ov-file#ideal-flowtemperature-calibration

**Comment by sb53systems** [comment]
Thanks for your feedback, I would have liked to diagnose this case, could you send me the generated G-Code.  
  
Salim.

**Comment by sb53systems** [comment]
An update will be posted in a few hours, which will solve many of your issues.  
Please let me know if you have any other tips.  
  
Salim

**Comment by sb53systems** [comment]
The negative temperature problem needs to be fixed, I'm trying to find its cause.  
  
here is the section of the code that is responsible for the temperature change  
![image](https://github.com/user-attachments/assets/4234560a-2a46-4a4d-8bae-a5fba13c1d66)  
  
Please feel free to help me if you have any programming knowledge.  
  
Salim.

**Comment by Nathan22211** [comment]
the only place I can think to look would be probably the getNextTemp math, that seems to be the only thing that can make a negitive. what I'd do is have a check in place to make sure nothing goes below the minimum temp, cus so far it seems there is no check there.

**Comment by Nathan22211** [comment]
yeah now I'm rather stumped. I'm not too familar with pascal syntax.

**Comment by Nathan22211** [comment]
it did only occur once so maybe it was a fluke? I hadn't had it occur again but I'll make sure to save the generated gcode next time it does

**Comment by sb53systems** [comment]
I think I was able to fix this problem here:  
![image](https://github.com/user-attachments/assets/aca01629-0604-459f-a2a3-6a59fbdb3f79)  
  
I was even able to avoid comments added in the same line.  
I'm able to use your G-Code ( https://github.com/sb53systems/G-Code-Flow-Temperature-Controller/issues/7#issuecomment-2391957604 )  
![image](https://github.com/user-attachments/assets/8b3cf738-d74d-440f-803b-2652fd89b4c4)

**Comment by sb53systems** [comment]
Could you send me another G-Code File ?

**Comment by sb53systems** [comment]
Here is the new updated file : https://github.com/sb53systems/G-Code-Flow-Temperature-Controller/blob/main/bin/SB53-Systems.exe  
Note that you need to replace this file only.  
Let me posted.  
  
Salim.

**Comment by sb53systems** [comment]
> I was even able to avoid comments added in the same line.  
  
![image](https://github.com/user-attachments/assets/67e867df-a6ac-447a-805f-e9464b5f5116)

**Comment by Nathan22211** [comment]
seems to work better. I will note that my temps can hardly keep up with the constant changes (Rapido v2 HF with CHT extension & nozzle) on my trident even at 15K accel, which is probably low considering it's 48V on the X, Y, and extruder. No clue if using Danger Klipper's MPC is making it worse or not, but it seems this script is more optimized for traditional PID on the extruder than MPC

**Comment by sb53systems** [comment]
Did you use the recommended settings in the Slicer? send me an example of a .3mf file.  
I couldn't try it on a printer other than mine, I'm waiting for your feedback and I'll try to help.  
  
How much do you rate the result on your printer?

**Comment by sb53systems** [comment]
If the temperature can't keep up, you just need to reduce the heat/cool down time in the script.  
Example : from 1c°/1s to 2c°/3s

**Comment by sb53systems** [comment]
Also note that this script is not specifically optimized for PID or MPC, you have to experiment yourself, and find the values ​​that suit. I think both are usable.  
  
I would be happy if people could share their experiences and settings, so that others can learn from them.  
  
I remain available to everyone if needed.  
  
Salim.
